### PR TITLE
travis: Post result to irc on failure

### PR DESCRIPTION
**Pull request by maflcko**
Replaces #11598 by @TheBlueMatt 

I am not entirely happy with my solution, as it will spam the irc with each failed job individually, instead one notice per travis build. However, assuming that very rarely more than one job per build fails, this might be acceptable.

### CODE REVIEW

**Comment by MarcoFalke** [comment]
Heh, please note that no notification is sent when the job times out.

**Review by promag — COMMENTED** [review]
Concept ACK.

Where is `irccat2.py`?

**Comment by laanwj** [comment]
Concept ACK

> However, assuming that very rarely more than one job per build fails, this might be acceptable.

If there's a bug, that will usually (unless it's platform specific) make all the jobs fails, or am I missing something?

**Comment by MarcoFalke** [comment]
Yes,

* this will only trigger on branches in the bitcoin/bitcoin repo, that is, pushes to branches or cron jobs running on branches.
* if there is a bug (let's say silent merge conflict in init.cpp), then all jobs will fail and all jobs will report error on irc. But merging such an obvious bug should happen never to rarely.

File: .travis.yml [hunk 1]
```diff
@@ -83,3 +83,7 @@ script:
 after_script:
     - echo $TRAVIS_COMMIT_RANGE
     - echo $TRAVIS_COMMIT_LOG
+after_failure:
+    - pip3 install irc --user
+    - if [[ "$TRAVIS_REPO_SLUG" = "bitcoin/bitcoin" && ( "$TRAVIS_EVENT_TYPE" = "push" || "$TRAVIS_EVENT_TYPE" = "cron" ) ]]; then wget "https://raw.githubusercontent.com/jaraco/irc/60f747a578157b03c952c050c765e2924d4fdc00/scripts/irccat2.py"; fi
```
Comment (laanwj) [inline .travis.yml]: Repeating this condition is a bit ugly. As this is YAML, isn't it possible to spread a script over multiple lines?

Comment (MarcoFalke) [reply .travis.yml]: Jup, that is possible

**Comment by MarcoFalke** [comment]
Closing, since it appears to be no longer working on travis, for me.

**Pull request closed without merging**